pkg/apis/azure/validation/secrets_test.go

```diff
@@ -39,86 +46,126 @@ var _ = Describe("Secret validation", func() {
 			map[string][]byte{
-				azure.SubscriptionIDKey: []byte("subscription"),
+				azure.SubscriptionIDKey: []byte(subscriptionID),
 				azure.TenantIDKey:       {},
-				azure.ClientIDKey:       []byte("cliendID"),
-				azure.ClientSecretKey:   []byte("clientSecret"),
+				azure.ClientIDKey:       []byte(clientID),
+				azure.ClientSecretKey:   []byte(clientSecret),
+			},
+			HaveOccurred(),
+		),
+
+		Entry("should return error when the tenant ID is not a valid UUID",
+			map[string][]byte{
+				azure.SubscriptionIDKey: []byte(subscriptionID),
+				azure.TenantIDKey:       append([]byte(" "), []byte(tenantID)...),
+				azure.ClientIDKey:       []byte(clientID),
+				azure.ClientSecretKey:   []byte(clientSecret),
 			},
 			HaveOccurred(),
 		),
 
 		Entry("should return error when the client ID field is missing",
 			map[string][]byte{
-				azure.SubscriptionIDKey: []byte("subscription"),
-				azure.TenantIDKey:       []byte("tenant"),
-				azure.ClientSecretKey:   []byte("clientSecret"),
+				azure.SubscriptionIDKey: []byte(subscriptionID),
+				azure.TenantIDKey:       []byte(tenantID),
+				azure.ClientSecretKey:   []byte(clientSecret),
 			},
 			HaveOccurred(),
 		),
 
 		Entry("should return error when the client ID is empty",
 			map[string][]byte{
-				azure.SubscriptionIDKey: []byte("subscription"),
-				azure.TenantIDKey:       []byte("tenant"),
+				azure.SubscriptionIDKey: []byte(subscriptionID),
+				azure.TenantIDKey:       []byte(tenantID),
 				azure.ClientIDKey:       {},
-				azure.ClientSecretKey:   []byte("clientSecret"),
+				azure.ClientSecretKey:   []byte(clientSecret),
+			},
+			HaveOccurred(),
+		),
+
+		Entry("should return error when the client ID is not a valid UUID",
+			map[string][]byte{
+				azure.SubscriptionIDKey: []byte(subscriptionID),
+				azure.TenantIDKey:       []byte(tenantID),
+				azure.ClientIDKey:       []byte("foo"),
+				azure.ClientSecretKey:   []byte(clientSecret),
 			},
 			HaveOccurred(),
 		),
 
 		Entry("should return error when the client secret field is missing",
 			map[string][]byte{
-				azure.SubscriptionIDKey: []byte("subscription"),
-				azure.TenantIDKey:       []byte("tenant"),
-				azure.ClientIDKey:       []byte("cliendID"),
+				azure.SubscriptionIDKey: []byte(subscriptionID),
+				azure.TenantIDKey:       []byte(tenantID),
+				azure.ClientIDKey:       []byte(clientID),
 			},
 			HaveOccurred(),
 		),
 
 		Entry("should return error when the client secret is empty",
 			map[string][]byte{
-				azure.SubscriptionIDKey: []byte("subscription"),
-				azure.TenantIDKey:       []byte("tenant"),
-				azure.ClientIDKey:       []byte("cliendID"),
+				azure.SubscriptionIDKey: []byte(subscriptionID),
+				azure.TenantIDKey:       []byte(tenantID),
+				azure.ClientIDKey:       []byte(clientID),
 				azure.ClientSecretKey:   {},
 			},
 			HaveOccurred(),
 		),
 
+		Entry("should return error when the client secret contains a trailing new line",
```

Comment:
Nit: for completeness you could add an entry that tests client secret containing a leading new line.

Reply:
I don't see much value in this, it's basically checking that `strings.Trim` behaves to specification. Then I would also have to check for `\r` etc. I would leave it out unless you insist.